journalpump/journalpump.py

```diff
@@ -542,6 +550,35 @@ def filter_fields(self, data):
         return {name: val for name, val in data.items() if (name.lstrip("_").lower() in self.fields) is self.whitelist}
 
 
+class UnitLogLevel:
+    def __init__(self, name, config):
+        self.name = name
+        self.levels = config
+        self.known_units = {}  # We stick units we have already run through the glob matcher here to avoid repeating it
+
+    def filter_by_level(self, data):
+        entry = data
+        if isinstance(data, bytes):
+            entry = json.loads(data.decode("utf-8"))
+        if not entry:
+            return {}
+        unit = entry.get("SYSTEMD_UNIT")
+        priority = entry.get("PRIORITY")
+        if ((not priority) or (not unit) or ((self.known_units.get(unit, False) or self._unit_match_level_glob(unit))
+                                             and priority <= LOG_SEVERITY_MAPPING[self.levels[unit]])):
+            self.known_units[unit] = True
+            return data
+        self.known_units[unit] = False
+        return {}
+
+    def _unit_match_level_glob(self, unit: str) -> bool:
+        for unit_pattern, level in self.levels.items():
```

Comment:
This order is semi-arbitrary (Python specifies dicts are sorted but you do not document it that way in the readme, and I think relying on it might be dangerous if there's e.g. json serialization/deserialization in the middle). I would recommend switching to a list to be more explicit, and it should be documented that it is first match that is used.